app/controllers/PayDateController.scala

```diff
@@ -112,4 +114,44 @@ class PayDateController @Inject()(
     } else {
       userAnswers.getList(PayDatePage).lift.apply(idx - 2)
     }
+
+  private def saveAndRedirect(userAnswers: UserAnswers, payDate: LocalDate, idx: Int): Future[Result] =
+    shouldGenerate(userAnswers, idx) match {
+      case (true, freq) =>
+        extractFurloughWithinClaimV(userAnswers) match {
+          case Valid(furlough) => {
+            val endDates = generateEndDates(freq, payDate, furlough)
+            for {
+              updatedAnswers <- Future.fromTry(userAnswers.setList(PayDatePage, endDates))
+              _              <- sessionRepository.set(updatedAnswers)
+            } yield Redirect(navigator.nextPage(PayDatePage, updatedAnswers, Some(idx)))
+          }
+          case Invalid(_) =>
+            Future.successful(Redirect(routes.ErrorController.somethingWentWrong()))
+        }
+      case (false, _) =>
+        for {
+          updatedAnswers <- Future.fromTry(userAnswers.setListItemWithInvalidation(PayDatePage, payDate, idx))
+          _              <- sessionRepository.set(updatedAnswers)
+        } yield Redirect(navigator.nextPage(PayDatePage, updatedAnswers, Some(idx)))
+    }
+
+  private def shouldGenerate(userAnswers: UserAnswers, idx: Int): (Boolean, PaymentFrequency) =
+    if (idx != 1) {
+      (false, Monthly)
```

Comment:
Could use arrowAssoc to prevent compiler warnings `a -> b` instead of `(a, b)`, as this requires auto-tupling.